### PR TITLE
Add support for leading vertical bar in union types

### PR DESCRIPTION
**Pull request by alexandrebodin**
Implementation of the leading vertical bar in the language parser.

Issue #897

### CODE REVIEW

File: src/language/__tests__/schema-printer-test.js [hunk 1]
```diff
@@ -83,6 +83,8 @@ union Feed = Story | Article | Advert
 
 union AnnotatedUnion @onUnion = A | B
 
+union AnnotatedUnionTwo @onUnion = A | B
```
Comment (robzhu) [inline src/language/__tests__/schema-printer-test.js]: Did you intend to have `| A | B`?

File: src/language/__tests__/schema-parser-test.js [hunk 2]
```diff
@@ -457,6 +457,28 @@ type Hello {
...
+            typeNode('Rld', { start: 21, end: 24 }),
+          ],
+          loc: { start: 0, end: 24 },
+        }
+      ],
+      loc: { start: 0, end: 24 },
+    };
+    expect(printJson(doc)).to.equal(printJson(expected));
+  });
+
```
Comment (robzhu) [inline src/language/__tests__/schema-parser-test.js]: Can you please also add test cases for `| Wo | Rld |`, `union Hello = Wo | Rld |`, and `union Hello = |`?

File: src/language/__tests__/schema-printer-test.js (hunk 1 above)
Comment (alexandrebodin) [inline src/language/__tests__/schema-printer-test.js]: Nope the printer will only print without it or with it I think it's better to leave it out and add it with prettier for ex

File: src/language/__tests__/schema-parser-test.js (hunk 2 above)
Comment (alexandrebodin) [inline src/language/__tests__/schema-parser-test.js]: Okay

**Comment by alexandrebodin** [comment]
@robzhu I added severals news tests

**Comment by robzhu** [comment]
@alexandrebodin, thank you for the high quality PR and fast turnaround!

File: src/language/parser.js [hunk 3]
```diff
@@ -943,6 +943,9 @@ function parseUnionTypeDefinition(lexer: Lexer<*>): UnionTypeDefinitionNode {
  */
 function parseUnionMembers(lexer: Lexer<*>): Array<NamedTypeNode> {
   const members = [];
+  if (peek(lexer, TokenKind.PIPE)) {
+    skip(lexer, TokenKind.PIPE);
+  }
```
Comment (leebyron) [inline src/language/parser.js]: The wrapping `peek` is unnecessary.  `skip` is effectively a call to `peek`.

File: src/language/parser.js [hunk 4]
```diff
@@ -943,6 +943,9 @@ function parseUnionTypeDefinition(lexer: Lexer<*>): UnionTypeDefinitionNode {
  */
 function parseUnionMembers(lexer: Lexer<*>): Array<NamedTypeNode> {
```
Comment (leebyron) [inline src/language/parser.js]: Every method in the parser has parse language atop it. It would be nice to include this here as well

**Comment by leebyron** [comment]
This is not the only place in the IDL where bar-separated tokens exists. Should we remain consistent and update the others as well? Or would that not be fitting?

**Comment by alexandrebodin** [comment]
@leebyron I could update the other places if you can point them to me.

**Comment by leebyron** [comment]
Let me take a quick look

**Pull request merged**